lib/kb_QualiMap/QualiMapRunner.py

```diff
@@ -56,19 +56,60 @@ def run_app(self, params):
         pprint(params)
         run_info = self.get_run_info(params)
 
-        if run_info['mode'] == 'single':
-            result = self.run_bamqc(params['input_ref'], run_info['input_info'])
-        elif run_info['mode'] == 'multi':
-            result = self.run_multi_sample_qc(params['input_ref'], run_info['input_info'])
-        else:
+        if run_info.get('mode') not in ['single', 'multi']:
             raise ValueError('Error in fetching the type to determine run settings.')
 
+        run_error = False
+        try:
+            if run_info['mode'] == 'single':
+                result = self.run_bamqc(params['input_ref'], run_info['input_info'])
+            elif run_info['mode'] == 'multi':
+                result = self.run_multi_sample_qc(params['input_ref'], run_info['input_info'])
+        except Exception as e:
+            run_error = True
+            result = {'qc_result_folder_path': None,
+                      'qc_result_zip_info': None,
+                      'shock_id': None}
+            error_msg = 'Running QualiMap with error:\n{}\n'.format(str(e))
```

Comment:
Say "returned error" instead of "with error"

Reply:
👍